File: Códigos beecrowd/torre_de_hanoi.c
#include <stdio.h>
#include <math.h>

int quadrado_perfeito(int soma) {

    double raiz = sqrt(soma);

    if(raiz == (int)raiz) return 1;
    else return 0;
}

int coloca_bolas(int n) {

    int hastes[n];
    for (int i = 0; i < n; i++) {
        hastes[i] = 0;
    }

    int bolas = 1, bolas_colocadas = 0;
    int i;

    for (i = 0; i < n; i++) {

        if (hastes[i] == 0) {
            hastes[i] = bolas;
            bolas++;
            bolas_colocadas++;
            i = -1;

        }else if (quadrado_perfeito(hastes[i] + bolas)){
            hastes[i] = bolas;
            bolas++;
            bolas_colocadas++;
            i = -1;
            
        }else if (i == n - 1){
            break;
        }

    }
    return bolas_colocadas;
}
 
int main() {
 
    int t, n;

    scanf("%d", &t);

    while(t > 0){
        scanf("%d", &n);

        int max_bolas = coloca_bolas(n);
    
        printf("%d\n", max_bolas);

        t--;
    }

    return 0;
}

File: Códigos beecrowd/torre_de_hanoi.py
import math


def quadrado_perfeito(soma) :
     print("Soma=",soma)
     raiz = math.isqrt(soma)
     print("Raiz=",raiz)
     
     return raiz * raiz == soma
     
     
def coloca_bolas(n):
     print("cheguei no coloca bolas")
     
     hastes =  [[] for _ in range(n)]
     bolas = 1
     bolas_colocadas = 0
     
     for haste in hastes:
         # tamanho = len(haste)
          if len(haste) == 0 :
               print("entrei no if 1")
               haste.append(bolas)
               bolas += 1
               bolas_colocadas += 1
          if len(haste) >= 1 :
               print("entrei no else")  
               if quadrado_perfeito(haste[-1] + bolas):
                    print("entrei no if 1 do quadrado perfeito")
                    haste.append(bolas)
                    bolas += 1
                    bolas_colocadas += 1
                    
          print("bolas=",bolas,"bolas_colocadas=",bolas_colocadas)
          print("vetor",haste)
        
     for i in hastes :    
          print(i)  
     return bolas_colocadas;
               

# num de casos teste
t = int(input())

while t > 0 :
    
     n = int(input())
    
     max_bolas = coloca_bolas(n)
     
     print(max_bolas)
    
     t -= 1